eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/SubmittedVariant.java

```diff
@@ -37,10 +39,11 @@
 
     private boolean supportedByEvidence;
 
-    public SubmittedVariant(String assemblyAccession, String taxonomyAccession,
-                            String projectAccession, String contig, long start,
-                            String referenceAllele, String alternateAllele,
-                            boolean supportedByEvidence) {
+    public SubmittedVariant(String assemblyAccession, int taxonomyAccession, String projectAccession, String contig,
+                            long start, String referenceAllele, String alternateAllele, boolean supportedByEvidence) {
+        Assert.hasText(assemblyAccession, "Assembly sccession is required");
+        Assert.hasText(projectAccession, "Project accession is required");
```

Comment:
We try to avoid adding a Spring dependency in models that do not require it. The same can be done throwing an `IllegalArgumentException`.